Sources/Sentry/SentryEnvelope.m

```diff
@@ -71,6 +71,67 @@ - (instancetype)initWithType:(NSString *)type
     return self;
 }
 
+- (NSDictionary *)serialize
+{
+
+    NSMutableDictionary *target = [[NSMutableDictionary alloc] init];
+    if (self.type) {
+        [target setValue:self.type forKey:@"type"];
+    }
+
+    if (self.filename) {
+        [target setValue:self.filename forKey:@"filename"];
+    }
+
+    if (self.contentType) {
+        [target setValue:self.contentType forKey:@"content_type"];
+    }
+
+    [target setValue:[NSNumber numberWithUnsignedInteger:self.length] forKey:@"length"];
+
+    return target;
+}
+
+@end
+
+@implementation SentryEnvelopeAttachmentHeader
+
+- (instancetype)initWithType:(NSString *)type length:(NSUInteger)length
+{
+    if (self = [super initWithType:type length:length]) {
+        _attachmentType = kSentryAttachmentTypeEventAttachment;
+    }
+    return self;
+}
+
+- (instancetype)initWithType:(NSString *)type
+                      length:(NSUInteger)length
+                    filename:(NSString *)filename
+                 contentType:(NSString *)contentType
+              attachmentType:(SentryAttachmentType)attachmentType
+{
+
+    if (self = [self initWithType:type length:length filenname:filename contentType:contentType]) {
+        _attachmentType = attachmentType;
+    }
+    return self;
+}
+
+- (NSDictionary *)serialize
+{
+    NSMutableDictionary *result =
+        [[NSMutableDictionary alloc] initWithDictionary:[super serialize]];
+    switch (self.attachmentType) {
```

Comment:
`m`: Instead of mapping the enum to type here, please add a method `nameForSentryAttachmentType`, and define constants for `event.view_hierarchy` and `event.attachment`, similar to what we have for `nameForSentrySpanStatus`.